lib/ansible/modules/packaging/os/redhat_repositories.py

```diff
@@ -0,0 +1,399 @@
+#!/usr/bin/python
+# vim:  et tw=120 ts=4 sw=4
+# Jerome Fenal (jfenal@redhat.com)
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: redhat_repository
+short_description: Manage repositories with RHSM using the C(subscription-manager) command
+description:
+    - Manage repositories with the Red Hat Subscription Management entitlement platform using the C(subscription-manager) command
+author: "Jerome Fenal (@jfenal)"
+notes:
+    - In order to be able to enable repositories, a system will
+      need first to be subscribed to RHSM. Use the
+      C(redhat_subscription) Ansible module for that matter.
+version_added: "2.3"
+requirements:
+    - subscription-manager
+options:
+    id:
+        description:
+            - I(RepoIDs) of repositories to enable or disable
+              When specifying multiple repos, separate them with a ",".
+              To specify all repositories, use "*".
+              In incremental mode, other repos won't be touched.
+              In idempotent mode (default), other repos would be disabled or enabled depending on their actual state.
+        required: False
+        default: null
+    state:
+        description:
+            - whether to enable (C(enabled)) or disable (C(disabled)) a repository
+        required: False
+        choices: [ "enabled", "disabled" ]
+        default: "enabled"
+    list:
+        description:
+            - List either all repositories, or only enabled or disabled
+        required: False
+        default: all
+        choices: [ "all", "enabled", "disabled" ]
+    mode:
+        description:
+            - Use either the idempotent or the incremental mode
+        required: False
+        default: "idempotent"
+        choices: [ "idempotent", "incremental" ]
+
+'''
+
+
+EXAMPLES = '''
+# List all repositories
+- redhat_repositories: list=all
```

Comment:
All examples should use block YAML stile instead of this Ansible inline style. The comment could form the tasks name. The above should be then changed to this: 

```
- name: List all repositories
  redhat_repositories:
    list: all
```

The same applies to all examples bellow.